core/templates/dev/head/components/SharingLinksDirective.js

```diff
@@ -45,6 +52,42 @@ oppia.directive('sharingLinks', [function() {
 
         $scope.escapedTwitterText = (
           oppiaHtmlEscaper.unescapedStrToEscapedStr($scope.getTwitterText()));
+
+        $scope.isNotDashboard = function() {
```

Comment:
Hi @anthkris -- I'm afraid this still leaks! The sharing-links directive still knows about the dashboard, and it shouldn't.

So, I took a pass at this; would you mind taking a look at what I've done in #2172? This shows how to do the refactor I was talking about, by passing exploration-id explicitly into the directive, and removing the use of explorationContextService. This makes sharing-links more reusable, and it also means that, when using sharing-links in the dashboard, you can now pass the exploration-id into it in the same way as in the other pages.)